src/main/java/io/confluent/connect/jdbc/util/QuoteMethod.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.connect.jdbc.util;
+
+public enum QuoteMethod {
+  ALWAYS("always"),
+  NEVER("never");
```

Comment:
Why are we using an enum as opposed to a boolean?  Do we imagine other `QuoteMethod` types in the future?  If so, maybe a comment?

Reply:
I did actually have `when_needed` as an alternative, but for this PR I wanted to keep things simple. Using a boolean type restricts future expansion, and I've been burned multiple times in the past when wanting to do that.

Reply:
If we start doing this in connectors in general (starting with an enum with values `always` and `never` for binary choices that we may want to expand later), it's a bit of cognitive overload to start to remember which configs should be `always`/`never` vs `true`/`false`.
Perhaps we could start with an enum with values `true` and `false` initially?
It seems more natural and easier to remember for me to say `quote.sql.identifier=true` or `quote.sql.identifier=false`, but that's probably just me.

Reply:
@rayokota, I do see your point, and unclear/unpredictable configues are all over our products. I will propose another solution -- a more suggestive config name. E.g. `quote.sql.identifiers` implies that the value should be a plurality of "identifiers," whatever those are. I think something like `sql.quote.method` more clearly suggests what type of thing you want. You've still got to read the docs to know the magic string "NEVER", but it's certainly clear it's not a boolean.

I think Randall does a nice thing by including this config in the sample configs of the repo, but I might suggest we consider a more holistic strategy of auto-generating sample configs with all configs and their commented-out documentation, but that'd be a conversation for another place.

Reply:
@cyrusv , I'm fine if the name of the config property naturally suggests there may be other possible values besides just a binary choice.  So in that case `quote.sql.identifiers` is not suggestive enough, I think `sql.quote.method` (as you propose) or `sql.quote.rule` are more suggestive that other possible config values may be added in the future.

Reply:
I don't think `quote.sql.method` is precise enough: this parameter does not affect all quoting of everything within a SQL statement -- string, date/time, etc. literal values are still always quoted. This only affects catalog/schema, table, and column _identifiers_.

I also dislike using `true` and `false` for values, because if we were to add other methods (e.g., `never`), the `true` and `false` values become incongruous.

Users will always have to review documentation to understand the constraints. And the REST API will expose `always` and `never` as the only valid values (at least currently). Any user using a tool that uses these allowed values will see them as options.

I agree that using boolean would be easier for now, but it locks us in if we want to add any future methods. So I prefer to keep this as-is.